### PR TITLE
IBX-2493: Added option to invite other users to DXP & register from invitation

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2493
| Versions      | 4.2

Added one line with `User:Invite policy` in [permissions.md](https://github.com/ezsystems/developer-documentation/pull/1685#diff-ddc82d90cd9a547cb0e8f6f53e92aaefd56e41d5fb9fee002cb0ed0ad5e2cb17)

- [x] Add invitations.md to TOC

### CODE REVIEW

File: docs/guide/permissions.md [hunk 1]
```diff
@@ -42,109 +42,110 @@ The more Role assignments and complex Policies you add for a given User, the mor
...
-||`lostorder_process`|process lost orders|
-||`orderhistory_view`|view Order history|
-||`quickorder`|access the quick order|
-||`read_basket`|see the basket|
-||`write_basket`|modify the basket (add, update, delete)|
-||`see_product_price`|see product prices in the catalog|
-|`siso_customercenter`</br>[[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]|`approve`|approve baskets in the customer center|
-||`buy`|buy as the customer center user|
-||`view`|access the customer center user management|
+| Module                                                                                                                | Function                                                | Effect                                                                                                                                                                                  |
```
Comment (DominikaK) [inline docs/guide/permissions.md:45]: Note: reformatting the table should not be included in this PR, because it's out of scope (and table fixes should cover all afflicted branches)

File: docs/api/public_php_api.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ The full list of available services covers:
 - [ContentService](https://github.com/ibexa/core/blob/main/src/contracts/Repository/ContentService.php)
 - [ContentTypeService](https://github.com/ibexa/core/blob/main/src/contracts/Repository/ContentTypeService.php)
 - [FieldTypeService](https://github.com/ibexa/core/blob/main/src/contracts/Repository/FieldTypeService.php)
+- [InvitationService](https://github.com/ibexa/user/blob/main/src/lib/Invitation/InvitationService.php)
```
Comment (ViniTou) [inline docs/api/public_php_api.md:31]:
```suggestion
- [InvitationService](https://github.com/ibexa/user/blob/main/src/contracts/Invitation/InvitationService.php)
```

File: docs/guide/user_management/invitations.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+```yaml
+# Invitation
+ibexa.site_access.config.default.user_invitation.templates.form: "@@IbexaUser/invitation/form.html.twig"
+ibexa.site_access.config.default.user_invitation.templates.mail: "@@IbexaUser/invitation/mail/user_invitation.html.twig"
+```
+
+For all those parameters, you must also set a SiteAccess, to which the new user will be invited to, for example site frontend:
+
+```yaml
+ibexa.site.config.default.user_invitation.templates.form: "@@IbexaUser/invitation/site_user_form.html.twig"
```
Comment (ViniTou) [inline docs/guide/user_management/invitations.md:38]: not exacly sure what you are trying to show here, cuz:
- you do not have to set those, you can, but if not it will fallback to default.
- those parameters name are wrong, it would be `ibexa.site_access.config.site.user_invitation.templates.mail` for `site` SA, not `ibexa.site.config.default.user_invitation.templates.mail`
- and there is a more friendly way for do that, with semantic confiuration : https://github.com/ibexa/user/blob/main/src/bundle/DependencyInjection/Configuration/Parser/UserInvitation.php

**Review by ViniTou — CHANGES_REQUESTED** [review]

File: docs/guide/permissions.md [hunk 4]
```diff
@@ -183,8 +183,8 @@ public function performAccessCheck(): void
 - `module` is the Policy module (e.g. `content`)
 - `function` is the function inside the module (e.g. `read`)
 - `limitations` are optional Limitations to check against. Here you can provide two keys:
-    - `valueObject` is the object you want to check for, for example `ContentInfo`.
-    - `targets` are a table of value objects that are the target of the operation.
+  - `valueObject` is the object you want to check for, for example `ContentInfo`.
```
Comment (MagdalenaZuba) [inline docs/guide/permissions.md:186]:
```suggestion
    - `valueObject` is the object you want to check for, for example `ContentInfo`.
```

File: docs/guide/permissions.md [hunk 5]
```diff
@@ -183,8 +183,8 @@ public function performAccessCheck(): void
...
+  - `targets` are a table of value objects that are the target of the operation.
```
Comment (MagdalenaZuba) [inline docs/guide/permissions.md:187]:
```suggestion
    - `targets` are a table of value objects that are the target of the operation.
```

File: docs/guide/user_management/invitations.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+To invite other members to the site or the Back Office User needs to have `User:Invite` permission added to his Role.
+You can limit the ability to invite other members to the specific User Groups
+e.g. Editors, or to the specific Roles within the group e.g. Admin, Buyer.
+
+## Creating and sending invitations
+
+Invitations are created with [InvitationService](https://github.com/ibexa/user/blob/main/src/lib/Invitation/InvitationService.php),
+but sending them requires additional setup.
+[[= product_name =]] provides you with `InvitationSender` interface for sending invitations via email.
```
Comment (ViniTou) [inline docs/guide/user_management/invitations.md:21]: We provide implementation of that interface. (\Ibexa\User\Invitation\MailSender)

File: docs/guide/user_management/invitations.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+## Expiring and refreshing invitations
+
+You can configure expiration time for invitation link in `user/src/bundle/Resources/config/ezplatform_default_settings.yaml` under Invitation section.
+
+```yaml
+# Invitation
+ibexa.site_access.config.default.user_invitation.hash_expiration_time: 'P2D'
+```
+If a User does not manage to click the invitation link sent to him, you can refresh the invitation.
+Refresh resets the time limit on sent link.
```
Comment (ViniTou) [inline docs/guide/user_management/invitations.md:75]: There is currently open PR that changes that behavior - refreshing invitation generates new date & new hash. 
https://github.com/ibexa/user/pull/46/commits

**Review by ViniTou — APPROVED** [review]

File: docs/guide/user_management/invitations.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+## Expiring and refreshing invitations
+
+You can configure expiration time for an invitation link in `user/src/bundle/Resources/config/ezplatform_default_settings.yaml` under an Invitation section.
+
+```yaml
+# Invitation
+ibexa.site_access.config.default.user_invitation.hash_expiration_time: 'P7D'
+```
+If a User does not manage to click the invitation link sent to him, you can refresh the invitation.
+Refresh resets the time limit and a hash in the invitation link.
```
Comment (DominikaK) [inline docs/guide/user_management/invitations.md:77]:
```suggestion
Refresh resets the time limit and changes the hash in the invitation link.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]
Remember to add the page to mkdocs.yml

File: docs/guide/user_management/invitations.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+                  mail: "@@App/invitation/mail.html.twig"
+ ```
+
+Here, you can specify which template should be used for the invitation mail,
+and what should be the expiration time for an invitation link included in that mail.
+
+## Configuration in YAML
+
+You can also configure invitation templates in `user/src/bundle/Resources/config/ezplatform_default_settings.yaml`.
+There, you can point to the specific forms and invitation email templates.
```
[the remark this answers is not in the document]
Comment (MagdalenaZuba) [reply docs/guide/user_management/invitations.md:51]: It is described in semantic configuration, you can do it both ways.

[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/user_management/invitations.md:51]: Then I'd rephrase it something like: default templates are configured in XYZ, but you can change them in (here show semantic config instead)

[the remark this answers is not in the document]
Comment (MagdalenaZuba) [reply docs/guide/user_management/invitations.md:51]: Semantic config is shown above

[the remark this answers is not in the document]
Comment (MagdalenaZuba) [reply docs/guide/user_management/invitations.md:51]: this is shown as additional setup

[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/user_management/invitations.md:51]: I'm still not sure about suggesting the user can change the values in ezplatform_default_settings.yaml, I don't believe they should. Double-check with devs, maybe?

File: docs/guide/user_management/invitations.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+```yaml
+ibexa.site_access.config.site.user_invitation.templates.form: "@@IbexaUser/invitation/site_user_form.html.twig"
+ibexa.site_access.config.site.user_invitation.templates.mail: "@@IbexaUser/invitation/mail/site_user_invitation.html.twig"
+```
+
+If the Site Access is not set, it will fall back to a default value.
+
+## Expiring and refreshing invitations
+
+You can configure expiration time for an invitation link in `user/src/bundle/Resources/config/ezplatform_default_settings.yaml` under an Invitation section.
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/user_management/invitations.md:70]: If the phrasing is changed above, here as well.

**Review by DominikaK — APPROVED** [review]
